pallets/collator-selection/src/lib.rs

```diff
@@ -162,6 +163,11 @@ pub mod pallet {
 
         /// The weight information of this pallet.
         type WeightInfo: WeightInfo;
+
+        /// The final word on whether the reported author can author at this height.
+        /// If the pallet that implements this trait depends on an inherent, that inherent **must**
+        /// be included before this one.
+        type CanAuthor: CanAuthor<Self::AccountId>;
```

Comment:
So in our concrete case the implementor is `AuthorInherent` which depends on `kick_off_authorship_validation` inherent. So `AuthorInherent` must be included before `CollatorSelection` in the `construct_runtime!` macro, is that it?

Reply:
It's somewhat explained [here](https://github.com/PureStake/nimbus/blob/ffb03c0614af1d3649fe52423ba99d98035baf74/pallets/author-inherent/src/lib.rs#L110) with context [here](https://github.com/paritytech/substrate/pull/10128#issuecomment-988237700)

author_inherent injects an inherent that is designed to run *after* all other inherents ( at least the ones nimbus' CanAuthor execution pipeline depends on itself ) but *before* extrinsics.

Inherents are executed in pallet inclusion order, so author_inherent must be the last one in line - in our case since we have `AllPalletsWithSystemReversed`, it must be listed before other pallets providing inherents.

**However**, this only affects *inherents* and neither Session nor CollatorSelection have any 🤡 
They all do their stuff in `on_initialize` which resolves before inherents are built, so in our case, pallet inclusion order does **not** matter.

I'll remove the other comment about sequencing from the code. Thanks for making me think on this a bit more

Reply:
> the implementor is AuthorInherent 

Actually, *every* pallet implementing `CanAuthor` is meant with this, so `author-inherent`, `manta-collator-selection` and `aura-style-filter` all need proper sequencing with other inherent providing pallets they depend on ( in our case: none )